ocaml/xapi/extauth_plugin_ADwinbind.ml

```diff
@@ -105,7 +192,45 @@ module AuthADWinbind : Auth_signature.AUTH_MODULE = struct
           explicitly filter any one-time credentials [like AD username/password for example] that it
           does not need long-term.]
   *)
-  let on_enable config_params = debug "on_enable To be implemented in CP-36085"
+  let on_enable config_params =
+    let user =
+      from_config ~name:"user" ~err_msg:"enable requires username"
+        ~config_params
+    in
+    let pass =
+      from_config ~name:"pass" ~err_msg:"enable requires password"
+        ~config_params
+    in
+
+    assert_hostname_valid () ;
+
+    let service_name =
+      (fun __context ->
+        Helpers.get_localhost ~__context |> fun host ->
+        Db.Host.get_external_auth_service_name ~__context ~self:host)
+      |> Server_helpers.exec_with_new_task
+           "retrieving external auth service name"
+    in
+    assert_domain_equal_service_name ~service_name ~config_params ;
+
+    let ou_config, ou_param = extract_ou_config ~config_params in
+
+    let workgroup = query_domain_workgroup service_name in
+    config_winbind_damon service_name workgroup ;
+
+    let args =
+      ["ads"; "join"; service_name; "-U"; user; "--no-dns-updates"] @ ou_param
+    in
+    debug "Joining domain %s with user %s and workgroup: %s" service_name user
+      workgroup ;
+    let env = [|Printf.sprintf "PASSWD=%s" pass|] in
```

Comment:
Please make sure this doesn't get logged, even on failure (on failure forkexecd likes to print its debug output, which might include env vars).

Reply:
The code does not print the env, https://github.com/xapi-project/forkexecd/blob/master/lib/forkhelpers.ml#L291
and I mocked a failure in my local dev environment, and It do not print the password on failure.

Reply:
you can write `Helpers.call_script ~log_output:Never ...` if you want to be doubly sure

Reply:
thanks @lippirk  for the help, but I maybe better to log the output on failure to help identify problems.